web/src/beta/features/WorkspaceSetting/innerPages/Members/Members.tsx

```diff
@@ -158,62 +135,85 @@ const Members: FC<Props> = ({
-      await handleUpdateMemberOfWorkspace({
+      const { status } = await handleUpdateMemberOfWorkspace({
         teamId: currentWorkspace?.id,
-        userId: user.id,
+        userId: member.userId,
         role: roleValue as Role
       });
-      setWorkspaceMembers((prevMembers) => {
-        return prevMembers.map((workspaceMember) =>
-          workspaceMember.id === user.id
-            ? {
-                ...workspaceMember,
-                role: roleValue as Role
-              }
-            : workspaceMember
-        );
-      });
+      if (status === "success") {
+        setCurrentWorkspace((prevMembers) => {
+          return {
+            ...prevMembers,
+            members: prevMembers?.members?.map((workspaceMember) =>
+              workspaceMember.userId === member.userId
+                ? {
+                    ...workspaceMember,
+                    role: roleValue as Role
+                  }
+                : workspaceMember
+            )
+          } as Workspace;
+        });
+      }
       setActiveEditIndex((prevIndex) => (prevIndex === index ? null : index));
     }
   };
 
-  const handleRemoveMemberButtonClick = (userId: string) => {
+  const handleRemoveMemberButtonClick = async (userId: string) => {
     if (currentWorkspace?.id) {
-      handleRemoveMemberFromWorkspace({
+      const { status } = await handleRemoveMemberFromWorkspace({
         teamId: currentWorkspace?.id,
         userId
       });
-      setWorkspaceMembers(
-        workspaceMembers.filter(
-          (workspaceMember) => workspaceMember.id !== userId
-        )
-      );
+      if (status === "success") {
+        setCurrentWorkspace((prevMembers) => {
+          return {
+            ...prevMembers,
+            members: prevMembers?.members?.filter(
+              (workspaceMember) => workspaceMember.userId !== userId
+            )
+          } as Workspace;
+        });
+      }
     }
   };
 
   const handleAddMember = () => {
-    memberSearchResults.forEach((memberSearchResult) => {
+    memberSearchResults.forEach(async (memberSearchResult) => {
       if (currentWorkspace?.id) {
-        handleAddMemberToWorkspace({
+        const { status } = await handleAddMemberToWorkspace({
           name: memberSearchResult.userName,
           teamId: currentWorkspace?.id,
           userId: memberSearchResult.id,
           role: Role.Reader
         });
-        setWorkspaceMembers((prevMembers) => [
-          ...prevMembers,
-          {
-            username: memberSearchResult.userName,
-            email: memberSearchResult.email,
-            role: Role.Reader,
-            id: memberSearchResult.id
-          }
-        ]);
+
+        if (status === "success") {
+          setCurrentWorkspace((prevMembers) => {
+            return {
+              ...prevMembers,
+              members: [
+                ...(prevMembers?.members || []),
+                {
+                  __typename: "TeamMember",
```

Comment:
Why do we need to define this??

Reply:
I did some refactor and this part is outdated, i'll close this conversation. Thank you!